clinica/pipelines/engine.py

```diff
@@ -607,8 +608,31 @@ def check_not_cross_sectional(self):
         from os import listdir
         from os.path import abspath, basename, dirname, isdir, join
 
+        from clinica.utils.exceptions import ClinicaInconsistentDatasetError
         from clinica.utils.stream import cprint
 
+        def detect_cross(all_subs, bids_dir, cross_subj, long_subj):
+            for sub in all_subs:
+                # Use is_cross_sectional to know if the current subject needs to be
+                # labeled as cross sectional
+                is_cross_sectional = False
+                folder_list = [
+                    f
+                    for f in listdir(join(bids_dir, sub))
+                    if isdir(join(bids_dir, sub, f))
+                ]
+                for fold in folder_list:
+                    if not fold.startswith("ses-"):
+                        is_cross_sectional = True
+                if is_cross_sectional:
+                    cross_subj.append(sub)
+                else:
+                    long_subj.append(sub)
```

Comment:
We can simplify this a little bit:

```python
for sub in all_subs:
    folder_list = ...
    if any([fold.startswith("ses-") for fold in folder_list]):
        cross_subj.append(sub)
    else:
        long_subj.append(sub)
```

Reply:
So, this actually doesn't work, because of it is reversed, it is actually the other way around.
Would something like this work ? It seems to work on my end.
```python
for sub in all_subs:
    folder_list = ...
    if not all([fold.startswith("ses-") for fold in folder_list]):
        cross_subj.append(sub)
    else:
        long_subj.append(sub)
```

Reply:
So, this actually doesn't work, because of it is reversed, it is actually the other way around.
Would something like this work ? It seems to work on my end.
```python
for sub in all_subs:
    folder_list = ...
    if not all([fold.startswith("ses-") for fold in folder_list]):
        cross_subj.append(sub)
    else:
        long_subj.append(sub)
```

Reply:
Yes ! 👍 Sorry for the wrong suggestion, I wrote this a bit too quickly...